src/Nethermind/Nethermind.Blockchain/TransactionPools/Filters/AcceptWhenTransactionFilter.cs

```diff
@@ -0,0 +1,202 @@
+using System;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Dirichlet.Numerics;
+
+namespace Nethermind.Blockchain.TransactionPools.Filters
+{
+    public class AcceptWhenTransactionFilter : ITransactionFilter
+    {
+        private readonly Builder.Filter _addFilter;
+        private readonly Builder.Filter _deleteFilter;
+
+        private AcceptWhenTransactionFilter(Builder.Filter addFilter, Builder.Filter deleteFilter)
+        {
+            _addFilter = addFilter;
+            _deleteFilter = deleteFilter;
+        }
+
+        public bool CanAdd(Transaction transaction) => Valid(transaction, _addFilter);
+        public bool CanDelete(Transaction transaction) => Valid(transaction, _deleteFilter);
```

Comment:
do we need to validate transactions when deleting them?

Reply:
The filter is dedicated to storage, so the question is: once transaction gets processed, and is removed from pending collection (internal for TxPool), should we also remove it from the storage, or apply a filter here?

Reply:
also remove from storage